src/main/java/ch/njol/skript/effects/EffHealth.java

```diff
@@ -53,47 +51,44 @@
 
 	static {
 		Skript.registerEffect(EffHealth.class,
-				"damage %livingentities/itemtypes% by %number% [heart[s]][ with fake cause %-damagecause%]",
-				"heal %livingentities% [by %-number% [heart[s]]]",
-				"repair %itemtypes% [by %-number%]");
+			"damage %livingentities/itemtypes% by %number% [heart[s]]",
+			"heal %livingentities% [by %-number% [heart[s]]]",
+			"repair %itemtypes% [by %-number%]");
 	}
 
-	@SuppressWarnings("null")
+	@SuppressWarnings("NotNullFieldNotInitialized")
 	private Expression<?> damageables;
 	@Nullable
 	private Expression<Number> damage;
 	private boolean heal = false;
-	@Nullable
-	private Expression<DamageCause> damageCause;
 
 	@SuppressWarnings({"unchecked", "null"})
 	@Override
-	public boolean init(final Expression<?>[] vars, final int matchedPattern, final Kleenean isDelayed, final ParseResult parser) {
-		damageables = vars[0];
-		if (ItemStack.class.isAssignableFrom(damageables.getReturnType())) {
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parser) {
+		damageables = exprs[0];
+		if (ItemType.class.isAssignableFrom(damageables.getReturnType())) {
 			if (!ChangerUtils.acceptsChange(damageables, ChangeMode.SET, ItemType.class)) {
 				Skript.error(damageables + " cannot be changed, thus it cannot be damaged or repaired.");
 				return false;
 			}
 		}
-		damage = (Expression<Number>) vars[1];
+		damage = (Expression<Number>) exprs[1];
 		heal = (matchedPattern >= 1);
-		
-		if (vars.length >= 3) damageCause = (Expression<DamageCause>) vars[2];
+
 		return true;
 	}
 
 	@Override
-	public void execute(final Event e) {
+	public void execute(Event e) {
 		double damage = 0;
 		if (this.damage != null) {
-			final Number n = this.damage.getSingle(e);
-			if (n == null)
+			Number number = this.damage.getSingle(e);
+			if (number == null)
 				return;
-			damage = n.doubleValue();
+			damage = number.doubleValue();
 		}
 		Object[] arr = damageables.getArray(e);
-		if (arr.length > 0 && arr[0] instanceof ItemType) {
+		if (ItemType.class.isAssignableFrom(damageables.getReturnType())) {
```

Comment:
Doesn't this mean the user won't be able to damage both an entity and an itemtype at the same time? While unlikely, there is no reason to remove the ability to do so. I'd opt for a for loop on the damageables expr and an instanceof check for each element instead